src/main/scala/where/Predicate.scala

```diff
@@ -1,10 +1,32 @@
 package com.originate.scalypher.where
 
 import com.originate.scalypher.ConstantString
+import com.originate.scalypher.path.Path
 import com.originate.scalypher.ToQuery
 
-sealed trait Predicate extends ToQuery
+case class PredicateWithProjection(
+  predicate: Predicate,
+  projection: PathProjection
+) {
+  def where(f: ObjectReference => Where): PredicateCondition =
```

Comment:
disregard, there isn't a distinction between what you can `where` on a `Node` vs a `Relationship` is there?